Sources/Build/BuildPlan.swift

```diff
@@ -701,11 +708,148 @@ public final class SwiftTargetBuildDescription {
+        result.append(buildParameters.toolchain.swiftCompiler.pathString)
+
+        result.append("-module-name")
+        result.append(target.c99name)
+        result.append("-incremental")
+        result.append("-emit-dependencies")
+
+        result.append("-output-file-map")
+        // FIXME: Eliminate side effect.
+        result.append(try! writeOutputFileMap().pathString)
+
+        if target.type == .library || target.type == .test {
+            result.append("-parse-as-library")
+        }
+        // FIXME: Handle WMO
+
+        result.append("-c")
+        for source in target.sources.paths {
+            result.append(source.pathString)
+        }
+
+        result.append("-I")
+        result.append(buildParameters.buildPath.pathString)
+
+        result += buildParameters.targetTripleArgs(for: target)
+        result += ["-swift-version", swiftVersion.rawValue]
+
+        result += buildParameters.indexStoreArguments
+        result += buildParameters.toolchain.extraSwiftCFlags
+        result += optimizationArguments
+        result += ["-g"]
+        result += ["-j\(ProcessInfo.processInfo.activeProcessorCount)"]
+        result += activeCompilationConditions
+        result += additionalFlags
+        result += moduleCacheArgs
+        result += buildParameters.sanitizers.compileSwiftFlags()
+        result += ["-parseable-output"]
+        result += self.buildSettingsFlags()
+        result += buildParameters.swiftCompilerFlags
+        return result
+    }
+
     /// Returns true if ObjC compatibility header should be emitted.
     private var shouldEmitObjCCompatibilityHeader: Bool {
         return buildParameters.triple.isDarwin() && target.type == .library
     }
 
+    private func writeOutputFileMap() throws -> AbsolutePath {
+        let path = tempsPath.appending(component: "output-file-map.json")
+        let stream = BufferedOutputByteStream()
+
+        stream <<< "{\n"
+
+        let masterDepsPath = tempsPath.appending(component: "master.swiftdeps")
+
+        stream <<< "  \"\": {\n";
+        // FIXME: Handle WMO
+        stream <<< "    \"swift-dependencies\": \"" <<< masterDepsPath.pathString <<< "\"\n";
+        stream <<< "  },\n";
+
+        // Write out the entries for each source file.
+        let sources = target.sources.paths
+        for (idx, source) in sources.enumerated() {
+            let object = objects[idx]
+            let objectDir = object.parentDirectory
+
+            let sourceFileName = source.basenameWithoutExt
+            let partialModulePath = objectDir.appending(component: sourceFileName + "~partial.swiftmodule")
+
+            let swiftDepsPath = objectDir.appending(component: sourceFileName + ".swiftdeps")
+
+            stream <<< "  \"" <<< source.pathString <<< "\": {\n"
+            // FIXME: Handle WMO
+            let depsPath = objectDir.appending(component: sourceFileName + ".d")
+            stream <<< "    \"dependencies\": \"" <<< depsPath.pathString <<< "\",\n"
```

Comment:
There's no existing abstraction for emitting the output file map in SwiftPM? Should we sink down [`OutputFileMap`](https://github.com/apple/swift-driver/blob/master/Sources/SwiftDriver/Driver/OutputFileMap.swift) from the Swift driver?

Reply:
Nope. Yeah, maybe we can bring that type in from the driver.